cclient/serialization/json/json_serializer.c

```diff
@@ -52,9 +148,44 @@ void json_get_inclusion_state_serialize_request(
     const serializer_t* const s, const get_inclusion_state_req_t* const obj,
     char* out) {}
 
-void json_get_inclusion_state_deserialize_response(
+int json_get_inclusion_state_deserialize_response(
     const serializer_t* const s, const char* const obj,
-    get_inclusion_state_res_t* out) {}
+    get_inclusion_state_res_t* out) {
+  cJSON* json_obj = cJSON_Parse(obj);
+  cJSON* json_item = NULL;
+
+  if (json_obj == NULL) {
+    cJSON_Delete(json_obj);
+    return EXIT_FAILURE;
+  }
+
+  json_item = cJSON_GetObjectItemCaseSensitive(json_obj, "error");
+  if (cJSON_IsString(json_item) && (json_item->string != NULL)) {
+    cJSON_Delete(json_obj);
+    return EXIT_FAILURE;
+  }
+
+  json_item = cJSON_GetObjectItemCaseSensitive(json_obj, "states");
+
+  if (cJSON_IsArray(json_item)) {
+    int int_items = cJSON_GetArraySize(json_item);
+    if (int_items <= 0) {
+      cJSON_Delete(json_obj);
+      return EXIT_FAILURE;
+    }
+  }
+
+  int int_cnt = 0;
+  cJSON* current_obj = NULL;
+  cJSON_ArrayForEach(current_obj, json_item) {
+    if (current_obj->string != NULL) {
+      out->bitmap.array[int_cnt] = strcmp("true", current_obj->string) ? 1 : 0;
```

Comment:
how do you know that `bitmap.array` is large enough? 
(also, why `bitmap`?)

Reply:
I'll fix it, the size of `bitmap.array` should refer to the set of tips and transactions in the request.
and rename to `states`